CRM/Upgrade/Incremental/php/FiveSixtyFive.php

```diff
@@ -40,6 +40,27 @@ public function upgrade_5_65_alpha1($rev): void {
     $this->addTask('Make Group.frontend_title required', 'alterColumn', 'civicrm_group', 'frontend_title', "varchar(255) NOT NULL COMMENT 'Alternative public description of the group.'", TRUE);
 
     $this->addTask('Update ActionSchedule.limit_to column', 'alterColumn', 'civicrm_action_schedule', 'limit_to', "int COMMENT 'Is this the recipient criteria limited to OR in addition to?'");
+    $this->addTask('Add fields to civicrm_mail_settings to allow more flexibility for email to activity', 'addMailSettingsFields');
+
+  }
+
+  public static function addMailSettingsFields() {
+    $alter = "ALTER TABLE civicrm_mail_settings
+      ADD `is_active` tinyint NOT NULL DEFAULT 1 COMMENT 'Ignored for bounce processing, only for email-to-activity',
+      ADD `activity_type_id` int unsigned COMMENT 'Implicit FK to civicrm_option_value where option_group = activity_type',
+      ADD `campaign_id` int unsigned DEFAULT NULL COMMENT 'Foreign key to the Campaign.',
+      ADD CONSTRAINT FK_civicrm_mail_settings_campaign_id FOREIGN KEY (`campaign_id`) REFERENCES `civicrm_campaign`(`id`) ON DELETE SET NULL,
+      ADD `activity_source` varchar(4) COMMENT 'Which email recipient to add as the activity source (from, to, cc, bcc).',
+      ADD `activity_targets` varchar(16) COMMENT 'Which email recipients to add as the activity targets (from, to, cc, bcc).',
+      ADD `activity_assignees` varchar(16) COMMENT 'Which email recipients to add as the activity assignees (from, to, cc, bcc).';";
+    CRM_Core_DAO::executeQuery($alter);
```

Comment:
This isn't re-run safe. The usual way to add columns is with the helper `addTask(...'addColumn'...)` - just `grep -r addColumn CRM/Upgrade` for examples.

Reply:
Thanks, have fixed.